rpc_jobs/hardening.yml

```diff
@@ -0,0 +1,133 @@
+- project:
+    name: 'RPC-Hardening-Jobs'
+    series:
+      - master:
+          branch: master
+          USER_VARS: ""
+    context:
+      - xenial:
+          DEFAULT_IMAGE: "16.04.2"
+    trigger:
+      - periodic:
+          CRON: "0 2 * * 7"
+    jobs:
+      - 'RPC-Hardening_{series}-{context}-{trigger}'
+
+- job-template:
+    name: 'RPC-Hardening_{series}-{context}-{trigger}'
+    project-type: workflow
+    properties:
+      - build-discarder:
+          days-to-keep: 7
+      - rpc-openstack-github
+    concurrent: true
+    triggers:
+      - timed: "{CRON}"
+    parameters:
+      - rpc_gating_params
+      - instance_params:
+          IMAGE: "OnMetal - Ubuntu 14.04 LTS (Trusty Tahr)"
+          FLAVOR: "onmetal-io1"
+          REGION: "IAD"
+      - rpc_repo_params:
+          RPC_BRANCH: "{branch}"
+      - osa_ops_params:
+          DEFAULT_IMAGE: "{DEFAULT_IMAGE}"
+          DATA_DISK_DEVICE: "sdb"
+      - text:
+          name: "USER_VARS"
+          default: "{USER_VARS}"
+          description: "OSA/RPC USER_VARS to inject for this build"
+      - string:
+          name: STAGES
+          default: |
+            Allocate Resources, Connect Slave, Prepare Multi-Node AIO,
+             Prepare RPC Configs, Deploy RPC w/ Script, Cleanup,
+             Destroy Slave
+          description: |
+            Pipeline stages to run CSV. Note that this list does not influence execution order.
+            Options:
+              Allocate Resources
+              Connect Slave
+              Prepare Multi-Node AIO
+              Prepare RPC Configs
+              Deploy RPC w/ Script
+              Prepare MaaS
+              Setup MaaS
+              Verify MaaS
+              Install Tempest
+              Tempest Tests
+              Prepare Horizon Selenium
+              Horizon Tests
+              Prepare Kibana Selenium
+              Kibana Tests
+              Holland
+              Pause (use to hold instance for investigation before cleanup)
+              Cleanup
+              Destroy Slave
+
+    dsl: |
+      node("CentOS"){{
+        try {{
+          dir("rpc-gating"){{
+            git branch: env.RPC_GATING_BRANCH, url: env.RPC_GATING_REPO
+            pubCloudSlave = load 'pipeline_steps/pubcloud.groovy'
+            ssh_slave = load 'pipeline_steps/ssh_slave.groovy'
+            common = load 'pipeline_steps/common.groovy'
+            multi_node_aio_prepare = load 'pipeline_steps/multi_node_aio_prepare.groovy'
+            deploy = load 'pipeline_steps/deploy.groovy'
+            maas = load 'pipeline_steps/maas.groovy'
+          }} // dir
+
+          withCredentials([
+            string(
+              credentialsId: 'hardening_repo_url',
```

Comment:
I'd prefer a parameter rather than a credential. Is it a cred because its an internal url?